File: netlab/imserver.py
#
# A messaging server in Python
#
# Name 1:
#
# Name 2:
#
#

import sys
import socket
from os import _exit as quit

def main():
    # parse arguments
    if len(sys.argv) != 2:
        print("usage: python3 %s <port>" % sys.argv[0])
        quit(1)
    port = sys.argv[1]
    
    # TODO 1: open a socket, bind, and listen

    # TODO 2: accept connections from client

    # message loop
    while(True):
        # TODO 3: receive a message from client and send response
        pass


if __name__ == "__main__":
    main()

File: netlab/imserver.c
/* 
 * A messaging server in C 
 *
 * Name 1:
 *
 * Name 2:
 *
 */ 

#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <arpa/inet.h>
#include <unistd.h>



int main(int argc, char **argv) {
  // parse arguments
  if (argc != 2) {
    fprintf(stderr, "usage: %s <port>\n", argv[0]);
    return 0;
  }
  unsigned port = atoi(argv[1]);
  
  // TODO 1: open socket, bind, and listen for incoming connections
  
  
  // TODO 2: accept incoming connection
  
  
  // message loop
  while(1){
    // TODO 3: receive message and send response
  }
  
  
  return 0;
}
